ddd-koans/Value Object & Entity/Draft/src/Notes.kt

```diff
@@ -0,0 +1,51 @@
+import java.math.BigDecimal
+import java.math.MathContext
+import java.time.Duration
+import NoteType.*
+
+data class Notes(private val notes: List<Note>) : Lengthy {
+    override fun lengthIn(tempo: Tempo): Length =
+        notes
+            .map { it.lengthIn(tempo) }
+            .reduce { acc, length -> acc + length }
+
+    operator fun plus(note: Note): Notes = Notes(notes + note)
+    operator fun plus(other: Notes): Notes = Notes(notes + other.notes)
+}
+
+data class Note(val noteType: NoteType, val pitch: Pitch) : Lengthy by noteType
+
+enum class NoteType(private val lengthInBeats: BigDecimal) : Lengthy {
```

Comment:
shouldn't it be named `NoteValue` instead of `NoteType`?

Reply:
Maybe we can somehow merge NoteValue with NoteType, or use NoteValue as the property of NoteType that would be used to count lengthInBeats? AFAIK, it would be enough to dive note value by 8 to obtain lengthInbeats?